.github/workflows/package_warnings_as_errors.yml

```diff
@@ -0,0 +1,25 @@
+name: Test-suite with warnings as error
+on:
+  # Scheduled trigger every Sunday at 2:35am UTC
+  schedule:
+  - cron: '35 2 * * 0'
+  workflow_dispatch:
+
+concurrency:
+  group: unit-tests-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  DURATIONS_BRANCH: bot/durations-update
+
+jobs:
+  test-warnings-as-errors:
+    uses: ./.github/workflows/interface-unit-tests.yml
+    secrets:
+      codecov_token: ${{ secrets.CODECOV_TOKEN }}
+    with:
+      branch: ${{ github.ref }}
+      pipeline_mode: 'warnings-as-errors'
```

Comment:
I don't see the `pipeline_mode` getting used anywhere. My suggestion is to get rid of the `python_warning_level` argument and only use the `pipeline_mode` you've added.

Reply:
I think I'll just remove pipeline section, but keep the setting via env-var as it allows us to also select different warning modes as required.